### DISCUSSION TITLE
Can replication be enabled for files already stored?

### DISCUSSION THREAD

**Discussion by tu-maurice** [seaweedfs/seaweedfs#1954] [Q&A] [answered]
Hi, I've been using seaweedfs for a few weeks. I have 3 master servers, 1 volume server, a filer and a s3 endpoint. I want to start up two more volume servers and enable replication. While I've found out how I would alter the default replication, which I presume will be used for future files, is there a way to retroactively replicate files that are already stored in the cluster?

**Comment by chrislusf** [comment, marked as answer]
You can use weed shell commands to change existing volume replication, and to fix replications.

**Reply by chrislusf** [reply]
In `weed shell`, run `volume.configure.replication` first, and then apply `volume.fix.replication` to add actual replicas.

**Reply by tu-maurice** [reply]
Sorry for the delay. I've tried to use volume.configure.replication but all it tells me is `error: empty replication value`

I've tried it without parameters, I've tried it with a replication like 001, I've tried to specify a volume id like `volume.configure.replication 15 001`. Everytime I get the same error of an empty replication value.

Also for some reason volume.list does not list show any volumes. Do I have to run weed shell somewhere particular like on the same host as the volume server?

Edit: I tried it on the volume server host and at least I got a list of all my volumes, but there's still the same error of an empty replication value.

**Reply by chrislusf** [reply]
`volume.configure.replication -h`

**Reply by tu-maurice** [reply]
I see. I only found `help volume.configure.replication` which did not mention those parameters. 

Either way. It worked perfectly once I knew how. Thank you very much!
